packages/pyright-internal/src/analyzer/typeEvaluator.ts

```diff
@@ -5359,7 +5360,7 @@ export function createTypeEvaluator(importLookup: ImportLookup, evaluatorOptions
                                 fileInfo.executionEnvironment.pythonVersion >= PythonVersion.V3_7 ||
                                 getAttrSymbol
                                     .getDeclarations()
-                                    .some((decl) => decl.path.toLowerCase().endsWith('.pyi'));
+                                    .some((decl) => decl.uri.extname.toLowerCase().endsWith('.pyi'));
```

Comment:
Is `extname...endsWith` correct? Is that necessary to handle foo.bar.pyi? Or is it superfluous?

Also makes me wonder if `Uri` should support a method like `hasFileExtension(".pyi")`

Reply:
endswith is superfluous here. Will fix.